server/app.js

```diff
@@ -165,13 +164,20 @@ const createApp = ({
     ),
   );
 
-  // GovUK Template Configuration
-  const establishmentId = getEstablishmentId(config.establishmentName);
-  app.locals.asset_path = '/public/';
-  app.locals.config = {
-    ...config,
-    establishmentId,
-  };
+  app.use((req, res, next) => {
+    const replaceUrl = /-prisoner-content-hub.*$/g;
+    config.establishmentName = pathOr('wayland', ['headers', 'host'], req)
```

Comment:
Would there be an issue with mutating the config? could we use a local variable?

Reply:
This would become a problem when merge the frontend instances in to a single deployment, mutating the config would affect other requests mid flight

Reply:
Then it needs a complete rethink :( Maybe establishmentName needs to be added into the req object? The requests need to know about the establishment name.

Reply:
This only ever seems to be used when setting the session so maybe we need to look at the ordering and check the session id? If it's the same session then we don't attempt to change the config? Or something?

Reply:
Or we put it in the session? Might make more sense? If it's not present then it must be a new session so we set the establishment.

Reply:
We could put it in the session, to avoid regenerating all the time. However it may cause issues, for when we switch between environments

Reply:
I think putting both `establishmentId` and `establishmentName` on the `req` object would work though

Reply:
They're already in the req object as part of config though :( Sessions shouldn't carry across environments as they're different sites?